compiler/rustc_transmute/src/layout/dfa.rs

```diff
@@ -277,3 +331,235 @@ where
         writeln!(f, "}}")
     }
 }
+
+use edge_set::EdgeSet;
+mod edge_set {
+    use std::cmp;
+
+    use run::*;
+
+    use super::*;
+    mod run {
+        use std::ops::{Range, RangeInclusive};
+
+        use super::*;
+        use crate::layout::Byte;
+
+        #[derive(Eq, PartialEq, Copy, Clone, Debug)]
+        pub(super) struct Run<S> {
+            // `start` and `end` are both inclusive (ie, closed) bounds, as this
+            // is required in order to be able to store 0..=255. We provide
+            // setters and getters which operate on closed/open ranges, which
+            // are more intuitive and easier for performing offset math.
+            start: u8,
+            end: u8,
+            pub(super) dst: S,
+        }
+
+        impl<S> Run<S> {
+            pub(super) fn new(range: RangeInclusive<u8>, dst: S) -> Self {
+                Self { start: *range.start(), end: *range.end(), dst }
+            }
+
+            pub(super) fn from_closed_open(range: Range<u16>, dst: S) -> Self {
```

Comment:
`from_inclusive_exclusive`, perhaps? that nomenclature is consistent with  https://doc.rust-lang.org/stable/std/ops/enum.Bound.html

Reply:
Done